### PR TITLE
Feature/test tools tests coverage

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

test_tools small improvements and new tests

## Fixes

- https://github.com/fetchai/agents-aea/issues/1464
- https://github.com/fetchai/agents-aea/issues/1411

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/test_tools/click_testing.py [hunk 1]
```diff
@@ -49,12 +49,12 @@ def invoke(
         color=False,
         **extra
     ):
-        """Patch click.testing.CliRunner.invoke()."""
+        """Call a cli command with click.testing.CliRunner.invoke."""
         exc_info = None
-        with self.isolation(input=input, env=env, color=color) as outstreams:
-            exception = None
-            exit_code = 0
```
Comment (DavidMinarsch) [inline aea/test_tools/click_testing.py:56]: @marcofavorito thoughts?

File: aea/test_tools/test_cases.py [hunk 2]
```diff
@@ -131,6 +133,7 @@ def force_set_config(cls, dotted_path: str, value: Any) -> None:
     def disable_aea_logging(cls):
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:133]: signature does not specify return type

File: aea/test_tools/test_cases.py [hunk 3]
```diff
@@ -405,40 +416,45 @@ def add_item(cls, item_type: str, public_id: str, local: bool = True) -> None:
         :return: None
         """
         cli_args = ["add", "--local", item_type, public_id]
-        if not local:
+        if not local:  # pragma: nocover
             cli_args.remove("--local")
-        cls.run_cli_command(*cli_args, cwd=cls._get_cwd())
+        return cls.run_cli_command(*cli_args, cwd=cls._get_cwd())
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:421]: return types do not match signature

File: aea/test_tools/test_cases.py [hunk 4]
```diff
@@ -405,40 +416,45 @@ def add_item(cls, item_type: str, public_id: str, local: bool = True) -> None:
...
+
         Run from agent's directory.
 
         :param item_type: str item type.
         :param name: name of the item.
 
         :return: None
         """
-        cls.run_cli_command("scaffold", item_type, name, cwd=cls._get_cwd())
+        return cls.run_cli_command("scaffold", item_type, name, cwd=cls._get_cwd())
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:435]: see above

File: aea/test_tools/test_cases.py [hunk 5]
```diff
@@ -405,40 +416,45 @@ def add_item(cls, item_type: str, public_id: str, local: bool = True) -> None:
...
+
         Run from agent's directory.
 
         :param item_type: str item type.
         :param name: public id of the item.
 
         :return: None
         """
-        cls.run_cli_command("fingerprint", item_type, public_id, cwd=cls._get_cwd())
+        return cls.run_cli_command(
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:449]: see above

File: aea/test_tools/test_cases.py [hunk 6]
```diff
@@ -447,29 +463,31 @@ def eject_item(cls, item_type: str, public_id: str) -> None:
...
     def run_install(cls):
         """
         Execute AEA CLI install command.
+
         Run from agent's directory.
 
         :return: None
         """
-        cls.run_cli_command("install", cwd=cls._get_cwd())
+        return cls.run_cli_command("install", cwd=cls._get_cwd())
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:477]: see above

File: aea/test_tools/test_cases.py [hunk 7]
```diff
@@ -447,29 +463,31 @@ def eject_item(cls, item_type: str, public_id: str) -> None:
...
         Generate AEA private key with CLI command.
+
         Run from agent's directory.
 
         :param ledger_api_id: ledger API ID.
 
         :return: None
         """
-        cls.run_cli_command("generate-key", ledger_api_id, cwd=cls._get_cwd())
+        return cls.run_cli_command("generate-key", ledger_api_id, cwd=cls._get_cwd())
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:490]: see above

File: aea/test_tools/test_cases.py [hunk 8]
```diff
@@ -489,15 +508,15 @@ def add_private_key(
         :return: None
         """
         if connection:
-            cls.run_cli_command(
+            return cls.run_cli_command(
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:511]: see above

File: aea/test_tools/test_cases.py [hunk 9]
```diff
@@ -514,28 +533,30 @@ def replace_private_key_in_file(
...
         Generate wealth with CLI command.
+
         Run from agent's directory.
 
         :param ledger_api_id: ledger API ID.
 
         :return: None
         """
-        cls.run_cli_command(
+        return cls.run_cli_command(
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:551]: see above

File: aea/test_tools/test_cases.py [hunk 10]
```diff
@@ -547,15 +568,16 @@ def get_wealth(cls, ledger_api_id: str = DEFAULT_LEDGER) -> str:
...
         """
         Replace the content of the source file to the dest file.
+
         :param src: the source file.
         :param dest: the destination file.
         :return: None
         """
         assert src.is_file() and dest.is_file(), "Source or destination is not a file."
-        src.write_text(dest.read_text())
+        dest.write_text(src.read_text())
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:580]: this changes the direction; check where the functionality is used...

File: aea/test_tools/test_cases.py [hunk 11]
```diff
@@ -771,15 +793,15 @@ class AEATestCaseMany(BaseAEATestCase):
...
 
     @classmethod
     def teardown_class(cls):
         """Teardown the test class."""
-        BaseAEATestCase.teardown_class()
+        super(AEATestCaseMany, cls).teardown_class()
 
 
-class AEATestCase(BaseAEATestCase):
+class AEATestCase(BaseAEATestCase):  # TODO: does not in use at the moment
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:804]: This is used in docs; please remove TODO

File: tests/test_packages/test_connections/test_p2p_libp2p/test_aea_cli.py [hunk 12]
```diff
@@ -68,7 +68,7 @@ def teardown_class(cls):
         """Tear down the test"""
         cls.terminate_agents()
 
-        AEATestCaseEmpty.teardown_class()
+        super(TestP2PLibp2pConnectionAEARunningDefaultConfigNode, cls).teardown_class()
```
Comment (DavidMinarsch) [inline tests/test_packages/test_connections/test_p2p_libp2p/test_aea_cli.py:71]: why is this needed?

File: aea/test_tools/test_cases.py (hunk 11 above)
Comment (solarw) [reply aea/test_tools/test_cases.py:804]: done

File: tests/test_packages/test_connections/test_p2p_libp2p/test_aea_cli.py (hunk 12 above)
Comment (solarw) [reply tests/test_packages/test_connections/test_p2p_libp2p/test_aea_cli.py:71]: incorrect call for classmethod.

AEATestCaseEmpty.teardown_class() - takes effect on AEATestCaseEmpty
not on current class instance.

File: aea/test_tools/test_cases.py (hunk 10 above)
Comment (solarw) [reply aea/test_tools/test_cases.py:580]: it used in documentation test, but it passes
also it looks like we should write src file content to destination file content not the opposite.

Comment (solarw) [reply aea/test_tools/test_cases.py:580]: can you check the logic i applied is ok?
